From e7647c4173b590ff1026d82ea7bd62956b53ecd9 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 15 May 2017 21:10:19 +0200
Subject: [PATCH] Added check to prevent image being 0x0 (reported in #489).

crafted file revealed an assertion failure in profile.c.
    magick: MagickCore/profile.c:1303: ResetImageProfileIterator: Assertion `image != (Image *) ((void *)0)' failed.

This fix CVE-2017-9141

origin: https://github.com/ImageMagick/ImageMagick/commit/f5910e91b0778e03ded45b9022be8eb8f77942cd
bug: https://github.com/ImageMagick/ImageMagick/issues/489
bug-debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863124
---
 coders/dds.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/coders/dds.c b/coders/dds.c
index 2f698dbb9..b22ad9930 100644
--- a/coders/dds.c
+++ b/coders/dds.c
@@ -1673,9 +1673,8 @@ static Image *ReadDDSImage(const ImageInfo *image_info,ExceptionInfo *exception)
   /*
     Initialize image structure.
   */
-  if (ReadDDSInfo(image, &dds_info) != MagickTrue) {
+  if (ReadDDSInfo(image, &dds_info) != MagickTrue)
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
-  }
 
   if (dds_info.ddscaps2 & DDSCAPS2_CUBEMAP)
     cubemap = MagickTrue;
@@ -1772,6 +1771,9 @@ static Image *ReadDDSImage(const ImageInfo *image_info,ExceptionInfo *exception)
   if (volume)
     num_images = dds_info.depth;
 
+  if (num_images < 1)
+    ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+
   for (n = 0; n < num_images; n++)
   {
     if (n != 0)
